analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -111,7 +111,8 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         var state = context.State;
 
         var operation = context.Operation.Instance;
-        if (operation.AsObjectCreation()?.Type.IsAny(TrackedCollectionTypes) ?? false)
+        if ((operation.AsObjectCreation()?.Type.IsAny(TrackedCollectionTypes) ?? false)
+            || IsEmptyArray(operation))
```

Comment:
I don't like the asymetry here too much. Changed `IsEmptyArray` is quite shorter, so it could get inlined

Reply:
#7367 will make it symmetrical, so I guess it's fine to keep it extracted

Reply:
I keep it for now. When merging #7367 we can make it one total.